saw-central/src/SAWCentral/Value.hs

```diff
@@ -747,7 +758,56 @@ getSharedContext :: TopLevel SharedContext
 getSharedContext = TopLevel_ (rwSharedContext <$> get)
 
 getJavaCodebase :: TopLevel JSS.Codebase
-getJavaCodebase = TopLevel_ (asks roJavaCodebase)
+getJavaCodebase =
+  do
+    status <- TopLevel_ (gets rwJavaCodebase)
+    case status of
+      JavaInitialized s -> pure s
+      JavaUnitialized   ->
+        do
+          opts <- getOptions
+          mb   <- liftIO (X.try (initJava opts))
+          case mb of
+            Right jcb ->
+              TopLevel_ (modify (\s -> s { rwJavaCodebase = JavaInitialized jcb })) >>
```

Comment:
`TopLevel` is also a `MonadState TopLevelRW` instance, so you ought to be able to get rid of the explicit newtype constructor here:

```suggestion
              modify (\s -> s { rwJavaCodebase = JavaInitialized jcb }) >>
```

Alternatively, you could use `modifyTopLevelRW`.